rezasm-source/rezasm-core/src/instructions/implementation/terminal_input_instructions.rs

```diff
@@ -0,0 +1,221 @@
+use crate::instructions::targets::input_target::InputTarget;
+use crate::simulation::reader::{Reader, ReaderBox};
+use crate::simulation::transform::transformable::Transformable;
+use crate::util::error::{IoError, SimulatorError};
+use crate::{
+    instruction,
+    instructions::{
+        instruction::Instruction,
+        instruction_registry::register_instruction,
+        targets::{input_output_target::InputOutputTarget, input_target::Input},
+    },
+    simulation::transform::transformation_sequence::TransformationSequence,
+    util::raw_data::RawData,
+};
+use lazy_static::lazy_static;
+use scanner_rust::ScannerAscii;
+
+lazy_static! {
+
+    /// Definition of the `readi` intruction, used to read an integer
+    pub static ref READI: Instruction =
+        instruction!(readi, |simulator: Simulator, output: InputOutputTarget| {
+            let mut scanner = ScannerAscii::new(simulator.get_reader_mut());
```

Comment:
@TrevorBrunette are you talking about declaring the scanner?

Reply:
Yeah, it's inefficient. We used a `OnceLock` on a regex that was repeatedly getting constructed. That approach doesn't seem viable here for borrow checker reasons. But we should try to figure something out.

Reply:
I don't believe it's an issue here. The function that this new writes is as follows:

```rust
impl<R: Read, N: ArrayLength + IsGreaterOrEqual<U4, Output = True>> ScannerAscii<R, N> {
    /// Create a scanner from a reader and set the buffer size via generics.
    ///
    /// ```rust
    /// use std::io;
    ///
    /// use scanner_rust::{generic_array::typenum::U1024, ScannerAscii};
    ///
    /// let mut sc: ScannerAscii<_, U1024> = ScannerAscii::new2(io::stdin());
    /// ```
    #[inline]
    pub fn new2(reader: R) -> ScannerAscii<R, N> {
        ScannerAscii {
            reader,
            buf: GenericArray::default(),
            buf_length: 0,
            buf_offset: 0,
            passing_byte: None,
        }
    }
}
```

which seems to be a near-zero-cost abstraction, especially since there is only one type of `R` per target.

Reply:
Additionally, removing this line will take some time as you'll need to restructure several objects to change them to `Scanner` types as opposed to `Box<Reader>` ones

Reply:
I am fairly sure that the scanner reads more into its buffer than it needs to. Letting it get dropped discards that data. Try running
```
readi $t0
readi $t1
```
and observe how, if two numbers are put on the same line, we ignore the second number. This *might* not be because of the scanners being constructed every time, but until it is proven to me that no remaining buffered characters are discarded, I would prefer a different approach used.

Reply:
Moving to its own bug. This is big enough as it is.